aesara/graph/basic.py

```diff
@@ -523,7 +528,8 @@ def clone(self):
         Tags and names are copied to the returned instance.
 
         """
-        cp = self.__class__(self.type, None, None, self.name)
+        name = name or self.name
+        cp = self.__class__(self.type, owner=None, index=None, name=name)
```

Comment:
This is causing the mypy flag in the tests. I thought the defined `OptionalApplyType` for owner arg would allow this. 
And it was throwing it either way... Has mypy changed at all?

Reply:
> Has mypy changed at all?

No, I think it's the other signature changes that make this an invalidly typed call for certain subclasses of `Variable`.

Reply:
What signature change are you talking about? 
The `__class__` in this case is Variable which didn't have a signature change (L437). Thought that would be what is used here.

Reply:
> What signature change are you talking about?
> The `__class__` in this case is `Variable` which didn't have a signature change ([L437](https://github.com/wd60622/aesara/blob/dda6490e5102c910656ebd4ebfc342077da54bfa/aesara/graph/basic.py#L434-L441)). Thought that would be what is used here.

The addition of signature changes in _subclasses_ of this class&mdash;at the time when that comments was made, of course.  For instance, `AtomicVariable.clone` currently seems problematic.

I think the current issue is likely due to the type information associated with `self.__class__`.  I'm guessing that this is inferred by Mypy to be a type that can't take a `None` value for its `owner` argument in the constructor.

There were always Mypy problems with the typing of the `owner` property and argument in `Variable`, and that led to the use of `OptionalApplyType` as a workaround, but I'm now assuming that this original issue is appearing again in a different form.

You can try using `Variable` instead of `self.__class__` and see if that removes the potentially conflicting `Generic` information from `OptionalApplyType`.  (N.B. Using `reveal_type(self.__class__)` might tell us what's going on.)